contracts/PaymentReceiver.sol

```diff
@@ -0,0 +1,63 @@
+pragma solidity ^0.8.11;
+
+import "hardhat/console.sol";
+import "./IPaymentReceiver.sol";
+
+contract ClientDatabase {
+    uint private nonce;
+    mapping (uint256 => address) private clientIds;
+    uint256[] private idList;
+    // An address cannot have more than 128 associated clients.
+    uint32 private maxClientIds = 128;
```

Comment:
What is the reason?

Reply:
we need to choose the size of the array that we put client ids into as we iterate over it (apparently dynamic arrays in memory are not allowed)